rust/lanzatool/src/esp.rs

```diff
@@ -1,7 +1,7 @@
 use anyhow::{Context, Result};
 use std::path::{Path, PathBuf};
 
-use crate::generation::Generation;
+use crate::generation::OSGeneration;
```

Comment:
How did you arrive at that name for the generation? My intuition would be to keep the name `Generation` in our code and use the fully qualified name if we refer to the bootspec generation (i.e. `bootspec::generation::Generation`) or even better change it upstream so we can just do `bootspec::Generation`

Reply:
So the intuition during the rename was:

- oh, name clashing, needs a new name
- this structure is about some "NixOS generation", combining: bootspec, specialisation name (if there's any) and version

While bootspec could theoretically be applied to any OS, this structure seems to be the NixOS-specific one, so I wanted to call it NixOSGeneration but the name was very ugly as you can imagine

But I have no strong feeling on this, we can keep generation and adopt the FQN

Reply:
`Generation` instead of `OsGeneration` would also make the rebase in my [confguration-limit branch](https://github.com/nix-community/lanzaboote/tree/configuration-limit) easier. I'd really appreciate it ;)

Reply:
Changed this back to `Generation` in 877a00e9 after speaking with @RaitoBezarius